DQMOffline/Trigger/plugins/TopMonitor.cc

```diff
@@ -575,17 +577,35 @@ void TopMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup
       edm::LogWarning("TopMonitor") << "Electron handle not valid \n";
       return;
   }
+  
+  //ATHER                                                                                                                                                                                                                       
+  edm::Handle<edm::ValueMap<bool> > eleIDHandle;
+  iEvent.getByToken(elecIDToken_,  eleIDHandle);
+  if (!eleIDHandle.isValid()){
+    edm::LogWarning("TopMonitor") << "Electron ID handle not valid \n";
+    return;
+  }
+
+
   std::vector<reco::GsfElectron> electrons;
   if (nelectrons_>0){
-      if ( eleHandle->size() < nelectrons_ ) return;
-      for ( auto const & e : *eleHandle ) {
-          if (eleSelection_(e)) electrons.push_back(e);
-          //Suvankar
-          if ( usePVcuts_ &&
-               (std::fabs(e.gsfTrack()->dxy(pv->position())) >= lepPVcuts_.dxy || std::fabs(e.gsfTrack()->dz(pv->position())) >= lepPVcuts_.dz) ) continue;
-      }
-      if ( electrons.size() < nelectrons_ ) return;
+    if ( eleHandle->size() < nelectrons_ ) return;
+    size_t index=0;                          //ATHER
+    for ( auto const & e : *eleHandle ) {
+      const auto el = eleHandle->ptrAt(index);  //ATHER                                                                                                                                                                                   
+      bool pass_id = (*eleIDHandle)[el];   //ATHER                                                                                                                                                                                         
+      index++;  //ATHER
+      if (eleSelection_(e) && pass_id) electrons.push_back(e); //ATHER
+      //Suvankar
+      if ( usePVcuts_ &&
```

Comment:
this line seems to have no effect?

Reply:
I think yes, but this is not my addition.